kyo-core/shared/src/main/scala/kyo/Scope.scala

```diff
@@ -0,0 +1,269 @@
+package kyo
+
+import kyo.Result.Error
+import kyo.Result.Panic
+import kyo.Tag
+import kyo.kernel.ContextEffect
+
+/** A structured effect for safe acquisition and finalization of resources.
+  *
+  * Scope provides a principled mechanism for working with entities that require proper cleanup, ensuring resources are released in a
+  * deterministic manner even in the presence of errors or interruptions. This effect is particularly valuable for managing external
+  * dependencies with lifecycle requirements such as file handles, network connections, database sessions, or any action that needs a
+  * corresponding cleanup step.
+  *
+  * Key features:
+  *   - Automatic resource finalization through `Scope.run` when computations complete or fail
+  *   - Compositional API allowing resource dependencies to be built up safely with `acquireRelease` and `acquire`
+  *   - Support for parallel cleanup through configurable concurrency levels with `run(closeParallelism)(...)`
+  *   - Declarative cleanup registration using `Scope.ensure` for custom finalizers
+  *   - Hierarchical scope management where child scopes are automatically closed when parent scopes close
+  *
+  * The Scope effect follows the bracket pattern (acquire-use-release) but with improved interruption handling and parallel cleanup
+  * capabilities. Scope finalizers registered with `ensure` are guaranteed to run exactly once when the associated scope completes, with
+  * failures in finalizers logged rather than thrown to avoid masking the primary computation result.
+  *
+  * Scope supports hierarchical nesting where child scopes are automatically closed when their parent scope closes. This ensures that
+  * resources acquired in child scopes are properly cleaned up even if the parent scope completes or fails.
+  *
+  * Typically, you would use `acquireRelease` to pair resource acquisition with its cleanup function, then compose multiple resources
+  * together before running the combined effect with `Scope.run`.
+  *
+  * @see
+  *   [[kyo.Scope.acquireRelease]] For creating resources with custom acquire and release functions
+  * @see
+  *   [[kyo.Scope.acquire]] For creating resources from Java Closeables
+  * @see
+  *   [[kyo.Scope.ensure]] For registering cleanup actions
+  * @see
+  *   [[kyo.Scope.run]] For executing resource-managed computations
+  */
+opaque type Scope <: Sync = Sync
+
```

Comment:
```suggestion
@deprecated("use `Scope`", "1.0-RC")
type Resource = Scope

@deprecated("use `Scope`", "1.0-RC")
val Resource = Scope
```

I think we could remove them after the 1.0

Reply:
That's fine, but can you note that in the message?